daktari/checks/docker.py

```diff
@@ -0,0 +1,57 @@
+import logging
+import re
+from typing import Optional
+
+from daktari.check import Check, CheckResult
+from daktari.command_utils import get_stdout
+from daktari.os import OS
+
+
+class DockerInstalled(Check):
+    name = "docker.installed"
+
+    suggestions = {OS.GENERIC: "Install docker: https://docs.docker.com/get-docker/"}
+
+    def check(self) -> CheckResult:
+        return self.verify_install("docker")
+
+
+major_version_pattern = re.compile("([0-9]+)")
+
+
+def get_major_docker_version() -> Optional[int]:
+    raw_version = get_stdout("docker version --format '{{.Server.Version}}'")
+    if raw_version:
+        match = major_version_pattern.search(raw_version)
+        if match:
+            major_version_string = match.group(1)
+            logging.debug(f"Docker major version: {major_version_string} (parsed from {raw_version})")
+            return int(major_version_string)
+    return None
+
+
+class DockerVersion(Check):
```

Comment:
I wonder if we can combine this into `DockerInstalled`, so you can say in your `.dakatari.py`:

```DockerInstalled(minimum_version = 19)``` (and the parameter is optional, so if the version check is only done if specified)